### PR TITLE
JenkinsQueueJobV2 - added handling of code 302 to JenkinsQueueJobV2 task 

### PR DESCRIPTION
**Pull request by anatolybolshakov**
**Task name**: JenkinsQueueJobV2

**Description**: reworked changes from https://github.com/microsoft/azure-pipelines-tasks/pull/16058 - fixed tests

**Documentation changes required:** Y

**Added unit tests:** added tests (302 code returned - considerCode302AsSuccess is on/off, simple successful scenario for task)

**Attached related issue:** N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected - made sanity check with simple jenkins job run

### CODE REVIEW

File: Tasks/JenkinsQueueJobV2/Tests/L0Consider302HttpCodeAsFailureByDefault.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    'osType': {
+        'osType': 'Windows'
+    }
+};
+tr.setAnswers(a);
+
+const requestClone: any = {};
+requestClone.get = function (params, callback: (err: any, httpResponse: any, body: string) => void) {
+    if (params.url === "bogusURL/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,%22:%22,//crumb)") {
```
Comment (KonstantinTyukalov) [inline Tasks/JenkinsQueueJobV2/Tests/L0Consider302HttpCodeAsFailureByDefault.ts:28]: What is this url?

File: Tasks/JenkinsQueueJobV2/Tests/L0SuccessfullSimpleRun.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    'osType': {
+        'osType': 'Windows'
+    }
+};
+tr.setAnswers(a);
+
+const requestClone: any = {};
+requestClone.get = function (params, callback: (err: any, httpResponse: any, body: string) => void) {
+    if (params.url === "bogusURL/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,%22:%22,//crumb)") {
```
Comment (KonstantinTyukalov) [inline Tasks/JenkinsQueueJobV2/Tests/L0SuccessfullSimpleRun.ts:28]: What is this url?

File: Tasks/JenkinsQueueJobV2/jenkinsqueuejobtask.ts [hunk 3]
```diff
@@ -113,12 +116,15 @@ export class TaskOptions {
 }
 
 async function doWork() {
+    let taskOptions: TaskOptions;
+
     try {
         tl.setResourcePath(path.join( __dirname, 'task.json'));
 
-        const taskOptions: TaskOptions = new TaskOptions();
+        taskOptions = new TaskOptions();
```
Comment (KonstantinTyukalov) [inline Tasks/JenkinsQueueJobV2/jenkinsqueuejobtask.ts:124]: Is that necessary change?

**Review by KonstantinTyukalov — APPROVED** [review]
LGTM, thanks! Please check comments

File: Tasks/JenkinsQueueJobV2/Tests/L0Consider302HttpCodeAsSuccess.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    },
+    'osType': {
+        'osType': 'Windows'
+    }
+};
+tr.setAnswers(a);
+
+const requestClone: any = {};
+requestClone.get = function (params, callback: (err: any, httpResponse: any, body: string) => void) {
+    if (params.url === "bogusURL/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,%22:%22,//crumb)") {
```
[the remark this answers is not in the document]
Comment (anatolybolshakov) [reply Tasks/JenkinsQueueJobV2/Tests/L0Consider302HttpCodeAsSuccess.ts:30]: Added comment, thanks!

File: Tasks/JenkinsQueueJobV2/Tests/L0SuccessfullSimpleRun.ts (hunk 2 above)
Comment (anatolybolshakov) [reply Tasks/JenkinsQueueJobV2/Tests/L0SuccessfullSimpleRun.ts:29]: Resolved the same way as above

**Comment by apatel69** [comment]
Is this still being worked on? Seeing similar issues to the original poster [here](https://github.com/microsoft/azure-pipelines-tasks/issues/15455)

**Pull request closed without merging**